File: gabut.py
import time
import sys

lyrics = [
    "Sungguh sayang aku tak bisa langsung mengungkapkan",
    "Perasaan yang kusimpan buatku tak tenang",
    "Ini semua karna hubungan pertemanan",
    "Kau sudah biasa anggapku sebagai kawan"
]

def type_out(text, delay=0.05):
    for char in text:
        sys.stdout.write(char)
        sys.stdout.flush()
        time.sleep(delay)
    print() 

if __name__ == "__main__":
    print("======== Mesrah Banget ========\n")
    time.sleep(3)

    for line in lyrics:
        type_out(line, delay=0.08) 
        time.sleep(0.5)      

File: gabut.cpp
#include <iostream>
#include <string>
#include <thread>   // untuk sleep
#include <chrono>   // untuk durasi

using namespace std;

void type_out(const string &text, int delay_ms = 80) {
    for (char c : text) {
        cout << c << flush; // flush biar langsung tampil per huruf
        this_thread::sleep_for(chrono::milliseconds(delay_ms));
    }
    cout << endl; // pindah baris setelah selesai
}

int main() {
    string lyrics[] = {
        "Sungguh sayang aku tak bisa langsung mengungkapkan",
        "Perasaan yang kusimpan buatku tak tenang",
        "Ini semua kar'na hubungan pertemanan",
        "Kau sudah biasa anggapku sebagai kawan"
    };

    cout << "Memulai lirik berjalan (efek ketik)..." << endl << endl;
    this_thread::sleep_for(chrono::seconds(1));

    for (const string &line : lyrics) {
        type_out(line, 80); // 80 ms per huruf (atur sesuai selera)
        this_thread::sleep_for(chrono::milliseconds(500)); // jeda antar baris
    }

    return 0;
}